src/main/java/com/hyperlocal/server/MerchantController.java

```diff
@@ -0,0 +1,92 @@
+package com.hyperlocal.server;
+
+import java.util.List;
+import java.util.Arrays;
+import java.util.concurrent.CompletableFuture;
+
+import com.github.jasync.sql.db.Connection;
+import com.github.jasync.sql.db.QueryResult;
+import com.github.jasync.sql.db.mysql.MySQLConnectionBuilder;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+
+
+// Rest Controller for Merchants
+
+@RestController
+public class MerchantController {
+  /* List of All merchants */
+
+  @GetMapping("/merchants/")
+  public CompletableFuture<List> getCatalog() {
+    Connection connection = MySQLConnectionBuilder.createConnectionPool(
+        "jdbc:mysql:///hyperlocal?socketFactory=com.google.cloud.sql.mysql.SocketFactory&cloudSqlInstance=speedy-anthem-217710:us-central1:hyperlocal");
+    CompletableFuture<List> future = CompletableFuture.supplyAsync(() -> {
+      CompletableFuture<QueryResult> queryResult = connection
+          .sendPreparedStatement("Select * from Merchants;");
+      return queryResult.join().getRows();
+    });
+    return future;
+  }
+
+  /*
+    Upserts details of a merchant
+      - MerchantID
+      - MerchantName
+      - MerchantPhone
+
+    Returns: Merchant Object with updated Details if exists, else return inserted Merchant object
+  */
+
+  @PostMapping("/upsert/merchants")
```

Comment:
@nagasrinivas-google we are taking merchant ID from client only till authentication is set up, right? It might lead to Insecure Direct Object Reference otherwise.